discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// metaLabelPrefix is the meta prefix used for all meta labels.
+// in this discovery.
+const (
+	metaLabelPrefix = model.MetaLabelPrefix + "scw_"
+	separator       = ","
+)
+
+// DefaultSDConfig is the default Scaleway Service Discovery configuration.
+var DefaultSDConfig = SDConfig{
+	Zone:            "fr-par-1",
+	Port:            80,
+	RefreshInterval: model.Duration(60 * time.Second),
+}
+
+type SDConfig struct {
+	// Project: The Scaleway Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple scaleway_sd_configs
+	Zone string `yaml:"zone"`
+	// Access Key
+	AccessKey string `yaml:"access_key"`
```

Comment:
Some product do requires access + secret key (example object storage) but you are right `instance` and `baremetal` products do not require access_key.
Usually, we recommend our users to always manipulate them as a pair as this is easier to debug for example access_key  can safely be printed into logs, communicated in a Github issue or support ticket.

Reply:
Since this is not needed, let's keep it out for simplicity. Additionally, a user could just put a wrong access key and it would still work, since unused.

Reply:
Removed

Reply:
I tested and access_key are required by the SDK. So they become mandatory here as well if there is no config file or environment variables.

Reply:
We could set it to `SCWNOACCESSTOKENNEED`, since it is not validated. I think it should be an implementation detail. The discover works well with that token.